lib/phoenix.ex

```diff
@@ -47,10 +47,33 @@ defmodule Phoenix do
     import Supervisor.Spec
 
     children = [
+      # Ensure format encoders are loaded before starting
+      worker(Task, [fn -> ensure_format_encoders() end], restart: :transient),
```

Comment:
I'm not thrilled about this approach but it's the only way to get reasonable error messages out of `start`. The downside is the default max_restarts shows the runtime error three times. Better ideas?

Reply:
Would making it a `temporary` worker solve this?

Reply:
Starting a worker sounds like too much.

@chrismccord if you are not happy with the error message when raising directly inside application, you have two options:

  1. Log and return `{:error, :missing_format_encoder}`
  2. Log and let the app boot anyway